tests/test_layouts.py

```diff
@@ -264,21 +266,26 @@ def testUMAP(self):
             )
 
         self.assertRaises(
-                ValueError, g.layout_umap,
+                InternalError, g.layout_umap,
                 sampling_prob=-0.01,
             )
 
-        self.assertRaises(
-                ValueError, g.layout_umap,
-                sampling_prob=1.01,
-            )
+        # TODO: this is currently accepted by the C core function...
+        # document at the Python level or fix at the C level?
```

Comment:
Invalid parameter values should be rejected at the C core level. I don't want to have to re-discover these problems and re-fix them in the Mathematica interface.

Reply:
agreed, let me fix it in C then